src/content/index.js

```diff
@@ -0,0 +1,268 @@
+	data = data.edges;
+	//
+	data = data.map((value) => RMPProfessorData.fromGraphQL(value.node));
+	return data;
+
+}
+
+
+///assumes that there is one prof data object provided
+function displaySingleProfRating(element, profData){
+
+	element.setAttribute('target', '_blank');
+	element.classList.add('blueText');
+	element.parentElement && element.parentElement.classList.add('classSearchBasicResultsText');
+
+
+	element.textContent += ` (${profData.getDifficultyRatingString()})`;
+	element.setAttribute('href', profData.getURL());
+
+	setupSingleProfToolTip(element, profData);
+}
+
+function setupSingleProfToolTip(element, profData) {
+	let wouldTakeAgainText;
+	let easyRatingText;
+
+	const div = document.createElement('div');
+	const title = document.createElement('div');
+	title.classList.add('prof-rating-title');
+	title.textContent = 'Rate My Professor Details';
+	div.appendChild(title);
+
+	const professorText = document.createElement('div');
+	professorText.classList.add('prof-rating-text');
+	professorText.textContent = `${profData.getFullName()}, Professor in ${profData.department}`;
+	div.appendChild(professorText);
+	const avgRatingText = document.createElement('div');
+	avgRatingText.classList.add('prof-rating-text');
+	avgRatingText.textContent = `Overall Quality: ${profData.getQualityRatingString("-")}/5`;
+	div.appendChild(avgRatingText);
+	const numRatingsText = document.createElement('div');
+	numRatingsText.classList.add('prof-rating-text');
+	numRatingsText.textContent = `Number of Ratings: ${profData.ratingsCount}`;
+	div.appendChild(numRatingsText);
+
+	
+
+	easyRatingText = document.createElement('div');
+	easyRatingText.classList.add('prof-rating-text');
+	easyRatingText.textContent = `Level of Difficulty: ${profData.difficultyGPA}`;
+	div.appendChild(easyRatingText);
+	wouldTakeAgainText = document.createElement('div');
+	wouldTakeAgainText.classList.add('prof-rating-text');
+	
+
+	wouldTakeAgainText.textContent = `Would take again: ${profData.wouldTakeAgainPercentage ?? "N/A"}%`;
+	div.appendChild(wouldTakeAgainText);
+	const topTagsText = document.createElement('div');
+	topTagsText.classList.add('prof-rating-text');
+	topTagsText.textContent = 'Top Tags: ';
+	if (profData.topTags.length > 0) {
+		topTagsText.textContent = profData.topTags.join(", ");
+		div.appendChild(topTagsText);
+	}
+	div.appendChild(document.createElement('br'));
+	
+	if (profData.mostHelpfulRating) {
+		const mostHelpfulReview = profData.mostHelpfulRating
+		const classText = document.createElement('div');
+		classText.classList.add('prof-rating-text');
+		classText.textContent = `Most Helpful Rating: ${mostHelpfulReview.course
+			}${mostHelpfulReview.isOnlineClass ? ' (Online)' : ''}`; // Mark if class was online
+		div.appendChild(classText);
+		const dateText = document.createElement('div');
+		dateText.classList.add('prof-rating-text');
+		dateText.textContent = mostHelpfulReview.date; //TODO: maybe convert to string date
```

Comment:
`dateText.textContent = new Date(mostHelpfulReview.date).toLocaleDateString(); //TODO: maybe convert to string date
`

Reply:
done!